lib/cutting_edge/app.rb

```diff
@@ -35,18 +35,37 @@ module CuttingEdge
   class App < Sinatra::Base
     include CuttingEdgeHelpers
 
+    set :views, ::File.join(::File.dirname(__FILE__), 'templates')
+    Tilt.register Tilt::ERBTemplate, 'html.erb'
+    set :public_folder, ::File.join(::File.dirname(__FILE__), 'public')
+
     logger filename: "#{settings.environment}.log", level: :trace
 
     before do
       @store = settings.store
     end
 
-    get %r{/(.+)/(.+)/(.+)/info} do |source, org, name|
+    get "/" do
+      @repos = CuttingEdge::App.repositories
+      erb :index
+    end
+
+    get %r{/(.+)/(.+)/(.+)/info/json} do |source, org, name|
       repo_defined?(source, org, name)
       content_type :json
       @store[@repo.identifier].merge({:language => @repo.lang}).to_json # Todo: check whether value exists yet? If not, call worker / wait / timeout?
     end
 
+    get %r{/(.+)/(.+)/(.+)/info} do |source, org, name|
+      repo_defined?(source, org, name)
+      @name = name
+      @svg = url("/#{source}/#{org}/#{name}/svg")
+      @colors = {ok: 'green', outdated_patch: 'yellow', outdated_minor: 'orange', outdated_major: 'red', unknown: 'gray'}
+      @specs = @store[@repo.identifier]
```

Comment:
We need to check whether this is nil and fail accordingly

Reply:
This is still necessary

Reply:
Now catching this in the view and displaying a message.